lib/opte/src/engine/port.rs

```diff
@@ -1441,17 +1443,33 @@ impl<N: NetworkImpl> Port<N> {
         res: &result::Result<ProcessResult, ProcessError>,
     ) {
         let flow_after = pkt.flow();
+
         cfg_if! {
             if #[cfg(all(not(feature = "std"), not(test)))] {
-                let flow_b_arg = flow_id_sdt_arg::from(flow_before);
-                let flow_a_arg = flow_id_sdt_arg::from(flow_after);
                 // XXX This would probably be better as separate probes;
                 // for now this does the trick.
-                let res_str = match res {
-                    Ok(v) => format!("{:?}", v),
-                    Err(e) => format!("ERROR: {:?}", e),
+                let (eb, extra_str) = match res {
+                    Ok(v @ ProcessResult::Drop { reason }) => (
+                        ErrorBlock::from_err(v),
+                        Some(format!("{reason:?}\0"))
+                    ),
+                    Ok(v) => (ErrorBlock::from_err(v), None),
```

Comment:
This is a bit confusing, we're in the `Ok` case but we're calling `ErrorBlock::from_err`?

Reply:
I think the type is maybe named confusingly at this point for what it can do  -- it was intended for `Err` values, but really any nested `enum` is what it's good at (and used for here). I'll rethink the naming.

Reply:
Okay, made some minor adjustments in 0edd7e6. If you think DError should also be renamed to suit its new multipurpose place, then we can do so.